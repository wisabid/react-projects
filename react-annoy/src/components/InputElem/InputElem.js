import React from 'react';

const InputElem = (props) => {
    let username;
    return (
        <>
            { props.focus === props.elid
                ? <legend>{props.fieldname?props.fieldname:props.elid}</legend>
                : null
            }   
            { props.loading
                ? <legend>Loading...</legend>
                : null
            }  
            <input autoComplete="off" id={props.elid} type={props.eltype} 
            placeholder={(props.focus === props.elid && props.eltype !== 'password')?'':props.placeholder} 
            onFocus={props.focusme} ref={(input) => username = input} onBlur={() => props.blurme(props.elid)}/> 
        </>
    )
}

export default InputElem;