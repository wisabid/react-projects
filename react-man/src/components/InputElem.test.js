import React from 'react';
import { shallow } from 'enzyme';
import InputElem from './InputElem';

describe('<inputElem />', () => {
    it('<InputElem /> should render correctly', () => {
        const component = shallow(<InputElem />);
        expect(component).toMatchSnapshot();
    })

})