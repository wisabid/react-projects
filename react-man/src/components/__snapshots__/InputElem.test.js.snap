// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<inputElem /> <InputElem /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputElem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <legend />,
        null,
        <input
          autoComplete="off"
          onBlur={[Function]}
          placeholder=""
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "legend",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "id": undefined,
          "onBlur": [Function],
          "onFocus": undefined,
          "placeholder": "",
          "type": undefined,
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <legend />,
          null,
          <input
            autoComplete="off"
            onBlur={[Function]}
            placeholder=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "legend",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "id": undefined,
            "onBlur": [Function],
            "onFocus": undefined,
            "placeholder": "",
            "type": undefined,
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<inputElem /> <inputElem /> should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <inputElem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "inputElem",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "inputElem",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
